src/simple_ble.rs

```diff
@@ -220,18 +169,52 @@ impl<CB: FnMut(usize, usize)> SubscribableCallback for BleCallback<CB> {
     }
 }
 
+pub struct ScanBuffer<'a> {
+    shared_memory: &'a [u8; BUFFER_SIZE_SCAN],
```

Comment:
Shouldn't this be `&'a mut...`?

Reply:
This is actually a shared buffer, hence you must be able to read from it. As it will be effectively owned by the handle of the share, you will have to either make the reference immutable or let the handle own the buffer and expose the buffer as immutable reference.

Reply:
Exactly, it is written by the driver and read in userland.

Reply:
But then the code violates Rust's safety guarantees. The conflict arises as you can pass a `buffer: &[u8]` to `BleDriver::start` but still modify the content of the array via the `unsafe` driver call. The solution would be to either transfer ownership of the array to `start` (But you will run into problems with unsized arrays) or you pass a `&mut [u8]'.

Reply:
The solution might be to split up start into a "share" and "subscribe" function and let the SharedMemory object borrow the buffer mutably. The buffer could then be exposed mutably by a private function and immutably by a public function and used by a callback by the latter.